Sources/Algorithms/Partition.swift

```diff
@@ -26,23 +26,23 @@ extension MutableCollection {
     subrange: Range<Index>,
     by belongsInSecondPartition: (Element) throws-> Bool
   ) rethrows -> Index {
-    if n == 0 { return subrange.lowerBound }
-    if n == 1 {
-      return try belongsInSecondPartition(self[subrange.lowerBound])
-        ? subrange.lowerBound
-        : subrange.upperBound
+    switch n {
+    case 0,
+         1 where try belongsInSecondPartition(self[subrange.lowerBound]):
```

Comment:
I personally think this switch is really hard to read. If we don't want the dangling ifs then I think guards could be a better option.

Reply:
I think this combination of the `0` and `1` cases is the issue for readability. The switch can stay, but let's move the `belongsInSecondPartition` test into the `1` branch.

Reply:
Fair enough.